src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -275,13 +289,11 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
         }
 
         if ($extra && $query = http_build_query($extra, '', '&', PHP_QUERY_RFC3986)) {
-            // "/" and "?" can be left decoded for better user experience, see
-            // http://tools.ietf.org/html/rfc3986#section-3.4
```

Comment:
I am for keeping / ? decoded  for better readability

Reply:
It's still there